### ISSUE TITLE
Naïve `endomorphism_algebra` implementation gets slow for large dimensions

### ISSUE THREAD

**Issue by mgkurtz** [thofma/Hecke.jl#1083]
For an algebra generated by several 50×50 matrices, say 40, computing the `endomorphism_algebra` consumes lots of memory and time. Precisely, time and memory goes into `vcat` and `right_kernel` (and there into `rref`) in the code snippet below:
https://github.com/thofma/Hecke.jl/blob/bc4f3c45f42a90fee9605d7ad2f7bb6b123df96b/src/AlgAssAbsOrd/Conjugacy/Conjugacy.jl#L567-L583
This code builds a 40⋅50²×50² “dense” matrix with ~only two~ 50+50 non-zero entries per row. Using sparse matrices instead of dense ones, and adding the rows iteratively to a rref instead of building a giant matrix, should reduce memory and time consumption ~by a lot~, I guess.

Additionally a comment, to explain the code, could help. I guess, the lines
https://github.com/thofma/Hecke.jl/blob/bc4f3c45f42a90fee9605d7ad2f7bb6b123df96b/src/AlgAssAbsOrd/Conjugacy/Conjugacy.jl#L593-L595
should be assertions, which (if they were assertions) already helped explaining the code.

**Comment by thofma** [comment]
Yes and yes. I only used it for toying around so far, but your suggestions sound good.

**Comment by mgkurtz** [comment]
Seems, like I am bad at counting. There are actually n+m (here 100) entries per row. So, even with sparse matrices that can take some time. Anyway, there are more efficient algorithms out there. I will see, how far I come, with the tools Hecke or Oscar already include.
